schema/proto3/gateway.proto

```diff
@@ -19,15 +19,18 @@ syntax = "proto3";
 package alcor.schema;
 
 option java_package = "com.futurewei.alcor.schema";
-option java_outer_classname = "AuxiliaryGateway";
+option java_outer_classname = "Gateway";
 
-enum AuxGatewayType {
+enum GatewayType {
     NONE = 0;
     ZETA = 1;
+    TGW = 2;
+    IGW = 3;
+    NGW = 4;
 }
 
-message AuxGateway {
-    AuxGatewayType aux_gateway_type = 1;
+message GatewayConfiguration {
+    GatewayType gateway_type = 1;
```

Comment:
since we moved Gateway as a top_level resource, we should add the following in the begining similar to other top level resources:

    uint32 revision_number = 1;

    string request_id = 2;
    string id = 3;        *** note that I move this up for consistancy
    UpdateType update_type = 4; // DELTA (default) or FULL

Reply:
more fields in the gateway contract are added.